tests/end_to_end_test/snapshot_fail_test.go

```diff
@@ -39,151 +40,158 @@ func TestSnapshotFail(t *testing.T) {
-			expectSuccess: map[os.FileMode]bool{
-				0000: false,
-				0100: false,
-				0400: false,
-				0500: true,
-			},
-		},
-		{
-			desc:        "Modify permissions of the snapshot source itself (source is an EMPTY directory)",
-			modifyEntry: filepath.Join(dir0Path, "dir1", "emptyDir2"),
-			snapSource:  filepath.Join(dir0Path, "dir1"),
-			expectSuccess: map[os.FileMode]bool{
-				0000: false,
-				0100: false,
-				0400: true,
-				0500: true,
-			},
-		},
-		{
-			desc:        "Modify permissions of a FILE in the snapshot directory",
-			modifyEntry: filepath.Join(dir0Path, "dir1", "file2"),
-			snapSource:  filepath.Join(dir0Path, "dir1"),
-			expectSuccess: map[os.FileMode]bool{
-				0000: true,
-				0100: true,
-				0400: true,
-				0500: true,
-			},
-		},
-		{
-			desc:        "Modify permissions of a DIRECTORY in the snapshot directory",
-			modifyEntry: filepath.Join(dir0Path, "dir1", "dir2"),
-			snapSource:  filepath.Join(dir0Path, "dir1"),
-			expectSuccess: map[os.FileMode]bool{
-				0000: false,
-				0100: false,
-				0400: false,
-				0500: true,
-			},
-		},
-		{
-			desc:        "Modify permissions of an EMPTY directory in the snapshot directory",
-			modifyEntry: filepath.Join(dir0Path, "dir1", "emptyDir2"),
-			snapSource:  filepath.Join(dir0Path, "dir1"),
-			expectSuccess: map[os.FileMode]bool{
-				0000: false,
-				0100: false,
-				0400: true,
-				0500: true,
-			},
-		},
-		{
-			desc:        "Modify permissions of a FILE in a subdirectory of the snapshot root directory",
-			modifyEntry: filepath.Join(dir0Path, "dir1", "file2"),
-			snapSource:  dir0Path,
-			expectSuccess: map[os.FileMode]bool{
-				0000: true,
-				0100: true,
-				0400: true,
-				0500: true,
-			},
-		},
-		{
-			desc:        "Modify permissions of a DIRECTORY in a subdirectory of the snapshot root directory",
-			modifyEntry: filepath.Join(dir0Path, "dir1", "dir2"),
-			snapSource:  dir0Path,
-			expectSuccess: map[os.FileMode]bool{
-				0000: false,
-				0100: false,
-				0400: false,
-				0500: true,
-			},
-		},
-		{
-			desc:        "Modify permissions of an EMPTY directory in a subdirectory of the snapshot root directory",
-			modifyEntry: filepath.Join(dir0Path, "dir1", "emptyDir2"),
-			snapSource:  dir0Path,
-			expectSuccess: map[os.FileMode]bool{
-				0000: false,
-				0100: false,
-				0400: true,
-				0500: true,
-			},
-		},
-	} {
-		t.Log(tc.desc)
-
-		uniqueSourceMap[tc.snapSource] = struct{}{}
-		restoreDir := fmt.Sprintf("%s%d", restoreDirPrefix, ti)
-		numSuccessfulSnapshots += testPermissions(e, t, tc.snapSource, tc.modifyEntry, restoreDir, tc.expectSuccess)
+	for _, ignoreFileErr := range []string{"true", "false"} {
+		for _, ignoreDirErr := range []string{"true", "false"} {
+			ignoringDirs := ignoreDirErr == "true"
+			ignoringFiles := ignoreFileErr == "true"
+
+			// Use "inherit" instead of "false" sometimes. Inherit defaults to false
+			if !ignoringFiles && rand.Intn(2) == 0 {
+				ignoreFileErr = "inherit"
```

Comment:
And the default in the parent policy is to NOT ignore the corresponding error, so the value of `ignoreFiles` still makes sense, is that correct?

Reply:
Right - at the bottom it defaults to not ignoring each, with ignore=false, set=true
```
var defaultErrorHandlingPolicy = ErrorHandlingPolicy{
	IgnoreFileErrors:         false,
	IgnoreFileErrorsSet:      true,
	IgnoreDirectoryErrors:    false,
	IgnoreDirectoryErrorsSet: true,
}
```